In [2]:
df = pd.read_csv("./data/Nepali_house_dataset.csv")

In [37]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

The history saving thread hit an unexpected error (OperationalError('attempt to write a readonly database')).History will not be written to the database.


In [38]:
df_clean = df.copy()

In [39]:
df['PRICE'] = df['PRICE'].astype(str)

In [40]:
df['PRICE'].unique()

array(['Rs. 2.9 Cr ', 'Rs. 4.75 Cr ', 'Rs. 1.99 Cr ', 'Rs. 4 Cr ',
       'Rs. 12000000', 'Rs. 27000000', 'Rs. 3.3 Cr ', 'Rs. 4.5 Cr ',
       'Rs. 3 Cr', 'Rs. 4.99 Cr ', 'Rs. 2.45 Cr ', 'Rs. 2.5 Cr ',
       'Rs. 4.25 Cr ', 'Rs. 6.5 Cr ', 'Rs. 3.6 Cr ', 'Rs. 5.8 Cr ',
       'Rs. 4.8 Cr ', 'Rs. 6.65 Cr ', 'Rs. 13.3 Cr ', 'Rs. 8.5 Cr ',
       'Rs. 5.5 Cr ', 'Rs. 2.95 Cr ', 'Rs. 6.35 Cr ', 'Rs. 10.5 Cr ',
       'Rs. 6.4 Cr ', 'Rs. 3.5 Cr ', 'Rs. 2.2 Cr ', 'Rs. 3.75 Cr ',
       'Rs. 8 Cr ', 'Rs. 7.75 Cr ', 'Rs. 16 Cr ', 'Rs. 3.19 Cr ',
       'Rs. 12 Cr ', 'Rs. 6.25 Cr ', 'Rs. 2.8 Cr ', 'Rs. 5.99 Cr ',
       'Rs. 3.35 Cr ', 'Rs. 3.50 Cr', 'Rs. 6 Cr ', 'Rs. 2.7 Cr ',
       'Rs. 2.75 Cr ', 'Rs. 11 Cr ', 'Rs. 3.25 Cr ', 'Rs. 3.1 Cr ',
       'Rs. 4 Cr', 'Rs. 3.15 Cr ', 'Rs. 4.65 Cr ', 'Rs. 5.95 Cr ',
       'Rs. 3.9 Cr ', 'Rs. 3.2 Cr ', 'Rs. 5.15 Cr ', 'Rs. 2.3 Cr ',
       'Rs. 1.7 Cr ', 'Rs. 2.4 Cr ', 'Rs. 2.25 Cr ', 'Rs. 1.8 Cr ',
       'Rs. 1.65 Cr ', 'Rs. 5.1 Cr ', 'Rs. 50 Lac/aa

# Starting Over

In [41]:
df['PRICE_CLEAN'] = df['PRICE'].astype(str).str.strip()

In [42]:
# Identify non-price rows first
non_price_mask = df['PRICE_CLEAN'].str.contains('call|contact|negotiable', na=False, case=False)
print(f"non price rows: {non_price_mask.sum()}")
print(df[non_price_mask]['PRICE'].unique())

non price rows: 190
['Price on call' 'Price on call ']


In [8]:
# Identify rate prices (which contains '/')
rate_mask = df['PRICE_CLEAN'].str.contains('/', na=False, regex=False)
print(f"Rates row: {rate_mask.sum()}")
print(df[rate_mask]['PRICE'].unique())
print(df[rate_mask]['PRICE'].head())


Rates row: 561
['Rs. 50 Lac/aana' 'Rs. 65,000 /m' 'Rs. 1.4 Lac/m' 'Rs. 1.5 Lac/m'
 'Rs. 1.05 Lac/m' 'Rs. 90 Lac/aana' 'Rs. 2.5 Lac/m' 'Rs. 70 Lac/aana'
 'Rs. 21 Lac/aana' 'Rs. 60,000 /m' 'Rs. 3.5 Lac/m' 'Rs. 80,000 /m'
 'Rs. 70,000 /m' 'Rs. 45,000 /m' 'Rs. 1.6 Lac/m' 'Rs. 1.28 Lac/m'
 'Rs. 90,000 /m' 'Rs. 1.7 Lac/m' 'Rs. 1.85 Lac/m' 'Rs. 3 Lac/m'
 'Rs. 1.55 Lac/m' 'Rs. 2 Lac/m' 'Rs. 82,000 /m' 'Rs. 1.3 Lac/m'
 'Rs. 1 Lac/m' 'Rs. 2.2 Lac/m' 'Rs. 5 Lac/m' 'Rs. 1.75 Lac/m'
 'Rs. 75,000 /m' 'Rs. 1.1 Lac/m' 'Rs. 4 Lac/m' 'Rs. 6 Lac/m'
 'Rs. 1.65 Lac/m' 'Rs. 2.25 Lac/m' 'Rs. 3.35 Lac/m' 'Rs. 4.5 Lac/m'
 'Rs. 1.25 Lac/m' 'Rs. 12.7 Lac/m' 'Rs. 2.75 Lac/m' 'Rs. 85,000 /m'
 'Rs. 8 Lac/m' 'Rs. 2.9 Lac/m' 'Rs. 50,000 /m' 'Rs. 78,900 /m'
 'Rs. 15 Lac/m' 'Rs. 2.64 Lac/m' 'Rs. 1.35 Lac/m' 'Rs. 3.25 Lac/m'
 'Rs. 1.2 Lac/m' 'Rs. 1.8 Lac/m' 'Rs. 55,000 /m' 'Rs. 2.3 Lac/m'
 'Rs. 42,000 /m' 'Rs. 3.1 Lac/m' 'Rs. 8.5 Lac/m' 'Rs. 7.25 Lac/m'
 'Rs. 3.85 Lac/m' 'Rs. 2.1 Lac/m' 'Rs. 53 Lac/aana' 'Rs. 1.15 Lac/m

In [9]:
# Identify absolute Cr prices (which donesnot contain '/', but has 'Cr')
cr_absolute_mask = (~rate_mask) & (~non_price_mask) & df['PRICE_CLEAN'].str.contains('Cr', na=False, case=False, regex=False)
print(f"Cr absolute rows: {cr_absolute_mask.sum()}")
print(df[cr_absolute_mask]['PRICE'].head())

Cr absolute rows: 2431
0     Rs. 2.9 Cr 
1    Rs. 4.75 Cr 
2    Rs. 1.99 Cr 
3       Rs. 4 Cr 
6     Rs. 3.3 Cr 
Name: PRICE, dtype: object


In [10]:
# Identify absolute Lac prices (no '/', has 'Lac', no 'Cr')
lac_absolute_mask = (~rate_mask) & (~non_price_mask) & (~cr_absolute_mask) & df['PRICE_CLEAN'].str.contains('Lac', na=False, case=False, regex=False)

print(f"Lac absolute rows: {lac_absolute_mask.sum()}")
print(df[lac_absolute_mask]['PRICE'].head())

Lac absolute rows: 28
210    Rs. 1.5 Lac 
357     Rs. 90 Lac 
386     Rs. 65 Lac 
546     Rs. 15 Lac 
642    Rs. 1.1 Lac 
Name: PRICE, dtype: object


In [11]:
# Identify raw NPR absolute prices (no '/', no 'Cr', no 'Lac', has numbers only)
raw_npr_mask = (~rate_mask) & (~non_price_mask) & (~cr_absolute_mask) & (~lac_absolute_mask)
print(f"Raw NPR absolute rows: {raw_npr_mask.sum()}")
print(df[raw_npr_mask]['PRICE'].head(5))

Raw NPR absolute rows: 208
4         Rs. 12000000
5         Rs. 27000000
140    Rs. 3,88,000,00
184     Rs. 27500000  
397     Rs. 27500000  
Name: PRICE, dtype: object


In [12]:
# NOw looking at rate rows only 

df_rate= df[rate_mask].copy()

# classifying rate rows by denominator
rate_aana_mask = df_rate['PRICE_CLEAN'].str.contains('/aana', na=False, case=False, regex=False)
rate_m_mask = df_rate['PRICE_CLEAN'].str.contains('/m', na=False, case=False, regex=False)

print(f"Rate /aana rows: {rate_aana_mask.sum()}")
print(f"Rate /m rows: {rate_m_mask.sum()}")
print(f"Rate other: {(~rate_aana_mask & ~rate_m_mask).sum()}")

print("\nRate /aana examples:")
print(df_rate[rate_aana_mask]['PRICE'].head(5))

print("\nRate /m examples:")
print(df_rate[rate_m_mask]['PRICE'].head(5))

Rate /aana rows: 26
Rate /m rows: 534
Rate other: 1

Rate /aana examples:
84      Rs. 50 Lac/aana
136     Rs. 90 Lac/aana
163     Rs. 70 Lac/aana
166     Rs. 21 Lac/aana
1019    Rs. 53 Lac/aana
Name: PRICE, dtype: object

Rate /m examples:
98      Rs. 65,000 /m
109     Rs. 1.4 Lac/m
111     Rs. 1.5 Lac/m
127    Rs. 1.05 Lac/m
159     Rs. 2.5 Lac/m
Name: PRICE, dtype: object


In [13]:
# finally verifying no overlap (all rows accounted)

total_accounted = (
    non_price_mask.sum() + 
    rate_mask.sum() + 
    cr_absolute_mask.sum() + 
    lac_absolute_mask.sum() + 
    raw_npr_mask.sum()
)

print(f"Total rows in original: {len(df)}")
print(f"Total rows classified: {total_accounted}")
print(f"Difference: {len(df) - total_accounted}")

if len(df) - total_accounted != 0:
    print("\nWARNING: Some rows not classified!")
    all_masks = non_price_mask | rate_mask | cr_absolute_mask | lac_absolute_mask | raw_npr_mask
    print(df[~all_masks]['PRICE'].head(10))

Total rows in original: 3418
Total rows classified: 3418
Difference: 0


# Finally all the classification have been done so i am moving to conversion

In [14]:
# Creating empty column for final NPR Values
df['PRICE_NPR'] = None

# Step1: Parse Cr absolute prices

In [15]:
# Identified which rows are Cr absolute (from earlier classification, which doesnot contain '/' but only 'Cr')
# Extracting just the number from PRICE_CLEAN THAT CONTAINS CR ABSOLUTE
# Example -> Rs. 2.9Cr -> 2.9 and then multiplying by 10000000
#.loc select rows and columns by labels
cr_numbers = df.loc[cr_absolute_mask, 'PRICE_CLEAN'].str.extract(r'(\d+\.?\d*)')[0].astype(float)
df.loc[cr_absolute_mask, 'PRICE_NPR'] = cr_numbers * 10000000
print(df[cr_absolute_mask][['PRICE', 'PRICE_NPR']].head(5))

          PRICE   PRICE_NPR
0   Rs. 2.9 Cr   29000000.0
1  Rs. 4.75 Cr   47500000.0
2  Rs. 1.99 Cr   19900000.0
3     Rs. 4 Cr   40000000.0
6   Rs. 3.3 Cr   33000000.0


# Step2: Parse Lac absolute prices

In [16]:
# alredy identified lac absolute rows
# extracting just jumbers from PRICE_CLEAN
# Rs. 1.5 Lac -> 1.5 and then multiplying by 100,000 to make it absolute number

lac_numbers = df.loc[lac_absolute_mask, 'PRICE_CLEAN'].str.extract(r'(\d+\.?\d*)')[0].astype(float)
df.loc[lac_absolute_mask, 'PRICE_NPR'] = lac_numbers * 100000
print(df[lac_absolute_mask][['PRICE', 'PRICE_NPR']].head(10))


             PRICE  PRICE_NPR
210   Rs. 1.5 Lac    150000.0
357    Rs. 90 Lac   9000000.0
386    Rs. 65 Lac   6500000.0
546    Rs. 15 Lac   1500000.0
642   Rs. 1.1 Lac    110000.0
683    Rs. 60 Lac   6000000.0
870     Rs. 3 Lac    300000.0
1097    Rs. 1 Lac    100000.0
1308   Rs. 85 Lac   8500000.0
1656   Rs. 85 Lac   8500000.0


# Step3: Parse Raw Npr absolute prices 

In [17]:
# actually we are converting  "Rs. 12000000" to float
# these are already in npr we just need to remove commas and convert into numbers
# already identified raw npr numbers 
# removing commas, then converting to float
# Example: "Rs. 3,65,00,000" -> remove commas -> "36500000" -> 36500000.0
raw_numbers = df.loc[raw_npr_mask, 'PRICE_CLEAN'].str.replace(',', '', regex=False).str.extract(r'(\d+)')[0].astype(float)
df.loc[raw_npr_mask, 'PRICE_NPR'] = raw_numbers

# Step5: Parse rate prices ( this is also connected with land area)

In [18]:
rate_rows = df[rate_mask].copy()
print(rate_rows['PRICE'].head(10))
print(rate_rows['PRICE'].unique()[:20])




84     Rs. 50 Lac/aana
98       Rs. 65,000 /m
109      Rs. 1.4 Lac/m
111      Rs. 1.5 Lac/m
127     Rs. 1.05 Lac/m
136    Rs. 90 Lac/aana
159      Rs. 2.5 Lac/m
163    Rs. 70 Lac/aana
166    Rs. 21 Lac/aana
168      Rs. 60,000 /m
Name: PRICE, dtype: object
['Rs. 50 Lac/aana' 'Rs. 65,000 /m' 'Rs. 1.4 Lac/m' 'Rs. 1.5 Lac/m'
 'Rs. 1.05 Lac/m' 'Rs. 90 Lac/aana' 'Rs. 2.5 Lac/m' 'Rs. 70 Lac/aana'
 'Rs. 21 Lac/aana' 'Rs. 60,000 /m' 'Rs. 3.5 Lac/m' 'Rs. 80,000 /m'
 'Rs. 70,000 /m' 'Rs. 45,000 /m' 'Rs. 1.6 Lac/m' 'Rs. 1.28 Lac/m'
 'Rs. 90,000 /m' 'Rs. 1.7 Lac/m' 'Rs. 1.85 Lac/m' 'Rs. 3 Lac/m']


In [19]:
# create columns for rate informations
df['RATE_VALUE'] = None # numeric rate value  ( 1.4 , 1.5 for 1.4 lac/m)
df['RATE_UNIT'] = None # unit for rate (lac or npr or cr)
df['RATE_DENOMINATOR'] = None # what it's per (m or aana)

# find rows that has /m (per meter)
rate_m = df[rate_mask & df['PRICE_CLEAN'].str.contains('/m', na=False)]
rate_aana = df[rate_mask & df['PRICE_CLEAN'].str.contains('/aana', na=False)]
rate_other = df[rate_mask & ~df['PRICE_CLEAN'].str.contains('/m|/aana', na=False)]
print(f"Rate /m rows: {len(rate_m)}")
print(f"Rate /aana rows: {len(rate_aana)}")
print(f"Rate other rows: {len(rate_other)}")

# Show examples of each
print("\nExamples of /m:")
print(rate_m['PRICE'].head(3))

print("\nExamples of /aana:")
print(rate_aana['PRICE'].head(3))

if len(rate_other) > 0:
    print("\nExamples of other:")
    print(rate_other['PRICE'].head(3))


Rate /m rows: 534
Rate /aana rows: 26
Rate other rows: 1

Examples of /m:
98     Rs. 65,000 /m
109    Rs. 1.4 Lac/m
111    Rs. 1.5 Lac/m
Name: PRICE, dtype: object

Examples of /aana:
84     Rs. 50 Lac/aana
136    Rs. 90 Lac/aana
163    Rs. 70 Lac/aana
Name: PRICE, dtype: object

Examples of other:
1583    Rs. 50 /sf
Name: PRICE, dtype: object


In [20]:
# Parse /m rows - (that has 'Lac')
#extract numbers, convert lac/m into npr

# first finding which /m rows have 'Lac'
m_with_lac = rate_m[rate_m['PRICE_CLEAN'].str.contains('Lac', na=False)]

#extract number and convert to NPR
extracted_numbers = df.loc[m_with_lac.index, 'RATE_VALUE'] = m_with_lac['PRICE_CLEAN'].str.extract(r'(\d+\.?\d*)')[0].astype(float)
converted_values = extracted_numbers * 100000

# set unit and denominator
df.loc[m_with_lac.index, 'RATE_VALUE'] = converted_values
df.loc[m_with_lac.index, 'RATE_UNIT'] = 'NPR'
df.loc[m_with_lac.index, 'RATE_DENOMINATOR'] = 'm'


In [21]:
# Parse /m rows - (that has no 'Lac')
# these are already npr, just need to remove comma just like before

m_without_lac = rate_m[~rate_m['PRICE_CLEAN'].str.contains('Lac', na=False, case=False)]
df.loc[m_without_lac.index, 'RATE_VALUE'] = m_without_lac['PRICE_CLEAN'].str.replace(',', '', regex=False).str.extract(r'(\d+)')[0].astype(float)
df.loc[m_without_lac.index, 'RATE_UNIT'] = 'NPR'
df.loc[m_without_lac.index, 'RATE_DENOMINATOR'] = 'm'


In [22]:
# Parse /anna rows - (all has 'Lac' so it might be easier then others)
#extrace number and convert to NPR
df.loc[rate_aana.index, 'RATE_VALUE'] = rate_aana['PRICE_CLEAN'].str.extract(r'(\d+\.?\d*)')[0].astype(float) * 100000
df.loc[rate_aana.index, 'RATE_UNIT'] = 'NPR'
df.loc[rate_aana.index, 'RATE_DENOMINATOR'] = 'aana'


In [23]:
#  Check all rate rows are parsed
total_rate_parsed = df['RATE_VALUE'].notna().sum()
expected_rate_rows = len(rate_m) + len(rate_aana)

print(f"Total rate rows parsed: {total_rate_parsed}")
print(f"Expected rate rows: {expected_rate_rows}")
print(f"Match: {total_rate_parsed == expected_rate_rows}")

# Show random sample of parsed rate rows
print("\nRandom sample of parsed rate rows:")
sample_rate = df[df['RATE_VALUE'].notna()].sample(50)
print(sample_rate[['PRICE', 'RATE_VALUE', 'RATE_UNIT', 'RATE_DENOMINATOR']])

Total rate rows parsed: 560
Expected rate rows: 560
Match: True

Random sample of parsed rate rows:
                PRICE RATE_VALUE RATE_UNIT RATE_DENOMINATOR
1448    Rs. 1.5 Lac/m   150000.0       NPR                m
1819    Rs. 4.5 Lac/m   450000.0       NPR                m
2101      Rs. 3 Lac/m   300000.0       NPR                m
1298    Rs. 2.5 Lac/m   250000.0       NPR                m
609     Rs. 1.4 Lac/m   140000.0       NPR                m
1523    Rs. 2.5 Lac/m   250000.0       NPR                m
895     Rs. 2.5 Lac/m   250000.0       NPR                m
2541    Rs. 1.5 Lac/m   150000.0       NPR                m
1719    Rs. 35,000 /m    35000.0       NPR                m
1262   Rs. 2.75 Lac/m   275000.0       NPR                m
711      Rs. 15 Lac/m  1500000.0       NPR                m
998     Rs. 1.5 Lac/m   150000.0       NPR                m
1357      Rs. 3 Lac/m   300000.0       NPR                m
2011   Rs. 1.75 Lac/m   175000.0       NPR                m


In [24]:
# Check what's ACTUALLY in PRICE_NPR column
print("Unique values in PRICE_NPR (first 20):")
print(df['PRICE_NPR'].head(20))

print("\nData types of first 5 values:")
for i in range(5):
    val = df['PRICE_NPR'].iloc[i]
    print(f"Row {i}: {val} -> type: {type(val)}")

Unique values in PRICE_NPR (first 20):
0     29000000.0
1     47500000.0
2     19900000.0
3     40000000.0
4     12000000.0
5     27000000.0
6     33000000.0
7     40000000.0
8     45000000.0
9     30000000.0
10    49900000.0
11    24500000.0
12    25000000.0
13    42500000.0
14    65000000.0
15    36000000.0
16    58000000.0
17    25000000.0
18    48000000.0
19    66500000.0
Name: PRICE_NPR, dtype: object

Data types of first 5 values:
Row 0: 29000000.0 -> type: <class 'float'>
Row 1: 47500000.0 -> type: <class 'float'>
Row 2: 19900000.0 -> type: <class 'float'>
Row 3: 40000000.0 -> type: <class 'float'>
Row 4: 12000000.0 -> type: <class 'float'>


In [25]:
# Convert PRICE_NPR to numeric (coerce errors to NaN)
df['PRICE_NPR'] = pd.to_numeric(df['PRICE_NPR'], errors='coerce')

# Convert RATE_VALUE to numeric
df['RATE_VALUE'] = pd.to_numeric(df['RATE_VALUE'], errors='coerce')

# Check dtypes again
print(df[['PRICE_NPR', 'RATE_VALUE']].dtypes)

PRICE_NPR     float64
RATE_VALUE    float64
dtype: object


In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3418 entries, 0 to 3417
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   TITLE             3418 non-null   object 
 1   LOCATION          3418 non-null   object 
 2   PRICE             3418 non-null   object 
 3   LAND AREA         3329 non-null   object 
 4   BUILDUP AREA      719 non-null    object 
 5   ROAD ACCESS       3409 non-null   object 
 6   FACING            3212 non-null   object 
 7   FLOOR             3323 non-null   float64
 8   BEDROOM           3136 non-null   float64
 9   BATHROOM          3072 non-null   float64
 10  BUILT YEAR        3357 non-null   object 
 11  PARKING           632 non-null    object 
 12  AMENITIES         3418 non-null   object 
 13  PRICE_CLEAN       3418 non-null   object 
 14  PRICE_NPR         2667 non-null   float64
 15  RATE_VALUE        560 non-null    float64
 16  RATE_UNIT         560 non-null    object 


In [27]:
# END OF PHASE 1 - SAVE CHECKPOINT
#df.to_parquet('./data/checkpoint_phase1_price_cleaned.parquet')

# Phase 2

In [1]:
import pandas as pd
import numpy as np
df = pd.read_parquet('./data/checkpoint_phase1_price_cleaned.parquet')

In [2]:
df.columns

Index(['TITLE', 'LOCATION', 'PRICE', 'LAND AREA', 'BUILDUP AREA',
       'ROAD ACCESS', 'FACING', 'FLOOR', 'BEDROOM', 'BATHROOM', 'BUILT YEAR',
       'PARKING', 'AMENITIES', 'PRICE_CLEAN', 'PRICE_NPR', 'RATE_VALUE',
       'RATE_UNIT', 'RATE_DENOMINATOR'],
      dtype='object')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3418 entries, 0 to 3417
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   TITLE             3418 non-null   object 
 1   LOCATION          3418 non-null   object 
 2   PRICE             3418 non-null   object 
 3   LAND AREA         3329 non-null   object 
 4   BUILDUP AREA      719 non-null    object 
 5   ROAD ACCESS       3409 non-null   object 
 6   FACING            3212 non-null   object 
 7   FLOOR             3323 non-null   float64
 8   BEDROOM           3136 non-null   float64
 9   BATHROOM          3072 non-null   float64
 10  BUILT YEAR        3357 non-null   object 
 11  PARKING           632 non-null    object 
 12  AMENITIES         3418 non-null   object 
 13  PRICE_CLEAN       3418 non-null   object 
 14  PRICE_NPR         2667 non-null   float64
 15  RATE_VALUE        560 non-null    float64
 16  RATE_UNIT         560 non-null    object 


In [4]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
FLOOR,3323.0,2.668492e+00,7.260466e-01,1.0,2.5,2.5,2.5,8.0
BEDROOM,3136.0,5.399235e+00,3.148436e+00,1.0,4.0,5.0,6.0,100.0
BATHROOM,3072.0,4.187826e+00,2.674617e+00,1.0,3.0,4.0,5.0,100.0
PRICE_NPR,2667.0,3.960263e+07,2.810322e+07,100000.0,24500000.0,32000000.0,45000000.0,400000000.0
RATE_VALUE,560.0,4.739730e+05,1.181845e+06,0.0,100000.0,180000.0,300000.0,9000000.0


In [5]:
df.head()

,TITLE,LOCATION,PRICE,LAND AREA,BUILDUP AREA,ROAD ACCESS,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_CLEAN,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 2.9 Cr,29000000.0,NaN,None,None
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",Rs. 4.75 Cr,47500000.0,NaN,None,None
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 1.99 Cr,19900000.0,NaN,None,None
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 4 Cr,40000000.0,NaN,None,None
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",Rs. 12000000,12000000.0,NaN,None,None


In [7]:
df[df['BEDROOM'] == 100][[ "LAND AREA","BEDROOM", "BATHROOM", "RATE_VALUE" , "LOCATION"]]


,LAND AREA,BEDROOM,BATHROOM,RATE_VALUE,LOCATION
2699,124.0 aana,100.0,100.0,1500000.0,"Imadol, Lalitpur"


In [12]:
df['LAND AREA'].unique()

array(['4.0 aana', '3.0 aana', '2.3 aana', '7.0 aana', '6.0 aana',
       '3.2 aana', '4.3 aana', '2.2 aana', '5.0 aana', '5.2 aana',
       '6.2 aana', '4.2 aana', '4.1 aana', '9.6 aana', '3.1 aana',
       '11.0 aana', '3.3 aana', '9.0 aana', '12 aana', '6.4 aana',
       '12.0 aana', '3.5 aana', '6 aana', '7.1 aana', '6.3 aana',
       '15.0 aana', '12.1 aana', '16.0 aana', '9.3 aana', '9.1 aana',
       '5.1 aana', '28.0 aana', '0.14 kattha', '24.0 aana', '14.4.0 aana',
       '13.2 aana', '32.0 aana', '0.0 aana', '6.1 aana', '2.9 aana',
       '7.2 aana', '8.0 aana', '10.2 aana', '35.0 aana', '37.0 aana',
       '10.0 aana', '127 sq. mtr', '22.0 aana', '30.0 aana', '18.0 aana',
       '23.0 aana', '7.3 aana', '14.0 aana', '13.0 aana', '1.0 aana',
       '5.3 aana', None, '40.0 aana', '2.5 aana', '0.12 kattha',
       '21.0 aana', '0.5 kattha', '1.2 aana', '8.2 aana', '20.0 aana',
       '1.1 aana', '3.8 aana', '1.9.9 kattha', '0.11 kattha', '9.2 aana',
       '2700 sq. ft', '48.0 

In [27]:
anna_count = df['LAND AREA'].str.contains('aana', na=False, case=False).sum()
aana_count = df['LAND AREA'].str.contains('aana', na=False, case=False).sum()
kattha_count = df['LAND AREA'].str.contains('kattha', na=False, case=False).sum()
sqft_count = df['LAND AREA'].str.contains('sq. ft', na=False, case=False).sum()
sqmtr_count = df['LAND AREA'].str.contains('sq. mtr', na=False, case=False).sum()
no_unit_count = df['LAND AREA'].notna() & ~df['LAND AREA'].str.contains('aana|kattha|sq|mtr', na=False, case=False, regex=False)

print(f"aana:     {aana_count} rows")
print(f"kattha:   {kattha_count} rows")
print(f"sq. ft:   {sqft_count} rows")
print(f"sq. mtr:  {sqmtr_count} rows")
print(f"no unit:  {no_unit_count.sum()} rows")
print(f"missing:  {df['LAND AREA'].isna().sum()} rows")

print(f"Total:    {len(df)} rows")

aana:     3251 rows
kattha:   39 rows
sq. ft:   33 rows
sq. mtr:  3 rows
no unit:  3329 rows
missing:  89 rows
Total:    3418 rows


In [34]:
# there are some wired deciman values too
weird_decimal = df[df['LAND AREA'].str.contains(r'\d+\.\d+\.', na=False)]
print(f"\nDouble decimals: {len(weird_decimal)}")
print(weird_decimal['LAND AREA'].head())



Double decimals: 3
163       14.4.0 aana
463      1.9.9 kattha
2215    0.9.25 kattha
Name: LAND AREA, dtype: object


In [35]:
no_unit = df[df['LAND AREA'].notna() & ~df['LAND AREA'].str.contains('aana|kattha|sq|mtr', na=False, case=False)]
print(f"\nNo unit specified: {len(no_unit)}")
print(no_unit['LAND AREA'].head())


No unit specified: 3
3213    9.25
3231     2.5
3326     4.2
Name: LAND AREA, dtype: object


In [46]:
print(" WEIRD ENTRIES")
weird_rows = df[df['LAND AREA'].str.contains(r'\.\d+\.', na=False)]
print(weird_rows[['LAND AREA', 'PRICE', 'LOCATION']])

 WEIRD ENTRIES
          LAND AREA            PRICE              LOCATION
163     14.4.0 aana  Rs. 70 Lac/aana      Nakkhu, Lalitpur
463    1.9.9 kattha      Rs. 2.5 Cr         Tandi, Chitwan
2215  0.9.25 kattha    Price on call  Devinagar, Rupandehi


In [49]:
no_unit_mask = (
    df['LAND AREA'].notna() & 
    ~df['LAND AREA'].str.contains('aana|kattha|sq|mtr', regex=True, na=False)
)
no_unit_rows = df[no_unit_mask] 
print(no_unit_rows[['LAND AREA', 'PRICE', 'LOCATION']])


     LAND AREA            PRICE                 LOCATION
3213      9.25  Rs. 7,35,00,000    Manamaiju,  Kathmandu
3231       2.5  Rs. 31,000,000    Kathmandu,  Sukedhara 
3326       4.2      Rs. 4.25 Cr      Raniban,  Kathmandu


In [51]:
def fix_double_decimal(val):
    if pd.isna(val):
        return val
    if str(val).count('.') > 1:
        parts = str(val).split('.')
        return f"{parts[0]}.{parts[1]} aana"
    return val

df['LAND AREA'] = df['LAND AREA'].apply(fix_double_decimal)

print(df[df['LAND AREA'] == '9.25'][['LOCATION', 'PRICE', 'LAND AREA']])


                   LOCATION            PRICE LAND AREA
3213  Manamaiju,  Kathmandu  Rs. 7,35,00,000      9.25


# Conversion Factors for Nepal:

1. **1 aana    = 1 aana** 
2. **1 kattha  = 20 aana**
3. **1 sq. ft  = 1/342.25 = 0.00292 aana**
4. **1 sq. mtr = 10.764/342.25 = 0.03145 aana**

In [53]:
# Create columns for number and unit
df['LAND_NUMBER'] = df['LAND AREA'].str.extract(r'(\d+\.?\d*)')[0].astype(float)
df['LAND_UNIT'] = 'unknown'

# Assign units based on text
df.loc[df['LAND AREA'].str.contains('aana', na=False), 'LAND_UNIT'] = 'aana'
df.loc[df['LAND AREA'].str.contains('kattha', na=False), 'LAND_UNIT'] = 'kattha'
df.loc[df['LAND AREA'].str.contains('sq. ft', na=False), 'LAND_UNIT'] = 'sqft'
df.loc[df['LAND AREA'].str.contains('sq. mtr', na=False), 'LAND_UNIT'] = 'sqmtr'

# Check result
print(df[['LAND AREA', 'LAND_NUMBER', 'LAND_UNIT']].head(20))

   LAND AREA  LAND_NUMBER LAND_UNIT
0   4.0 aana          4.0      aana
1   3.0 aana          3.0      aana
2   2.3 aana          2.3      aana
3   7.0 aana          7.0      aana
4   6.0 aana          6.0      aana
5   6.0 aana          6.0      aana
6   3.2 aana          3.2      aana
7   4.3 aana          4.3      aana
8   2.2 aana          2.2      aana
9   5.0 aana          5.0      aana
10  5.2 aana          5.2      aana
11  3.0 aana          3.0      aana
12  4.0 aana          4.0      aana
13  4.0 aana          4.0      aana
14  6.2 aana          6.2      aana
15  4.2 aana          4.2      aana
16  5.0 aana          5.0      aana
17  3.2 aana          3.2      aana
18  5.2 aana          5.2      aana
19  4.1 aana          4.1      aana


In [55]:
# Create final column in aana
df['LAND_AREA_AANA'] = df['LAND_NUMBER'].copy()

# Convert kattha to aana (multiply by 20)
kattha_mask = df['LAND_UNIT'] == 'kattha'
df.loc[kattha_mask, 'LAND_AREA_AANA'] = df.loc[kattha_mask, 'LAND_NUMBER'] * 20

# Convert sqft to aana (divide by 342.25)
sqft_mask = df['LAND_UNIT'] == 'sqft'
df.loc[sqft_mask, 'LAND_AREA_AANA'] = df.loc[sqft_mask, 'LAND_NUMBER'] / 342.25

# Convert sqmtr to aana (multiply by 10.764 then divide by 342.25)
sqmtr_mask = df['LAND_UNIT'] == 'sqmtr'
df.loc[sqmtr_mask, 'LAND_AREA_AANA'] = df.loc[sqmtr_mask, 'LAND_NUMBER'] * 10.764 / 342.25

# Check results

print(df[df['LAND_UNIT'] != 'aana'][['LAND AREA', 'LAND_NUMBER', 'LAND_UNIT', 'LAND_AREA_AANA']].head(20))

       LAND AREA  LAND_NUMBER LAND_UNIT  LAND_AREA_AANA
138  0.14 kattha         0.14    kattha        2.800000
228  127 sq. mtr       127.00     sqmtr        3.994238
332         None          NaN   unknown             NaN
342  0.12 kattha         0.12    kattha        2.400000
343  0.12 kattha         0.12    kattha        2.400000
357   0.5 kattha         0.50    kattha       10.000000
464  0.11 kattha         0.11    kattha        2.200000
480         None          NaN   unknown             NaN
484  2700 sq. ft      2700.00      sqft        7.888970
505         None          NaN   unknown             NaN
531         None          NaN   unknown             NaN
546   1.7 kattha         1.70    kattha       34.000000
548         None          NaN   unknown             NaN
558   3.0 kattha         3.00    kattha       60.000000
559  0.10 kattha         0.10    kattha        2.000000
563         None          NaN   unknown             NaN
608         None          NaN   unknown         

In [57]:
# Quick statisticshuge = df[df['LAND_AREA_AANA'] == df['LAND_AREA_AANA'].max()]
print(huge[['LAND AREA', 'LAND_AREA_AANA', 'PRICE', 'LOCATION', 'TITLE']])
print(f"Total rows with land area: {df['LAND_AREA_AANA'].notna().sum()}")
print(f"Min: {df['LAND_AREA_AANA'].min():.2f} aana")
print(f"Max: {df['LAND_AREA_AANA'].max():.2f} aana")
print(f"Mean: {df['LAND_AREA_AANA'].mean():.2f} aana")
print(f"Median: {df['LAND_AREA_AANA'].median():.2f} aana")

# Check for outliers
print(f"\nProperties > 50 aana: {(df['LAND_AREA_AANA'] > 50).sum()}")
print(f"Properties > 100 aana: {(df['LAND_AREA_AANA'] > 100).sum()}")

Total rows with land area: 3329
Min: 0.00 aana
Max: 2233.81 aana
Mean: 8.00 aana
Median: 4.20 aana

Properties > 50 aana: 34
Properties > 100 aana: 13


In [58]:
huge = df[df['LAND_AREA_AANA'] == df['LAND_AREA_AANA'].max()]
print(huge[['LAND AREA', 'LAND_AREA_AANA', 'PRICE', 'LOCATION', 'TITLE']])

            LAND AREA  LAND_AREA_AANA        PRICE           LOCATION  \
2122  2233.81 sq aana         2233.81  Rs. 2.8 Cr   Balkot, Bhaktapur   

                                     TITLE  
2122  Newly built house for sale in Balkot  


In [59]:
print(huge[[ "LAND AREA","BEDROOM", "BATHROOM", "RATE_VALUE" , "LOCATION"]])

            LAND AREA  BEDROOM  BATHROOM  RATE_VALUE           LOCATION
2122  2233.81 sq aana      4.0       4.0         NaN  Balkot, Bhaktapur


In [60]:
# This is clearly an error, set to NaN
df.loc[2122, 'LAND_AREA_AANA'] = None
df.loc[2122, 'LAND_UNIT'] = 'error'

In [62]:
df['LAND_UNIT'].unique()

array(['aana', 'kattha', 'sqmtr', 'unknown', 'sqft', 'error'],
      dtype=object)

In [63]:
df.head()

,TITLE,LOCATION,PRICE,LAND AREA,BUILDUP AREA,ROAD ACCESS,FACING,FLOOR,BEDROOM,BATHROOM,...,PARKING,AMENITIES,PRICE_CLEAN,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_NUMBER,LAND_UNIT,LAND_AREA_AANA
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,...,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 2.9 Cr,29000000.0,NaN,None,None,4.0,aana,4.0
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,...,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",Rs. 4.75 Cr,47500000.0,NaN,None,None,3.0,aana,3.0
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,...,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 1.99 Cr,19900000.0,NaN,None,None,2.3,aana,2.3
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,...,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",Rs. 4 Cr,40000000.0,NaN,None,None,7.0,aana,7.0
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,...,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",Rs. 12000000,12000000.0,NaN,None,None,6.0,aana,6.0


In [64]:
# since now land area are all in aana, i can safely remove land number and land unit that was only created to seperate and convert all unites to same one

df = df.drop(['LAND_NUMBER', 'LAND_UNIT'], axis=1)
print(df.columns.tolist())


['TITLE', 'LOCATION', 'PRICE', 'LAND AREA', 'BUILDUP AREA', 'ROAD ACCESS', 'FACING', 'FLOOR', 'BEDROOM', 'BATHROOM', 'BUILT YEAR', 'PARKING', 'AMENITIES', 'PRICE_CLEAN', 'PRICE_NPR', 'RATE_VALUE', 'RATE_UNIT', 'RATE_DENOMINATOR', 'LAND_AREA_AANA']


In [66]:
df = df.drop(['PRICE_CLEAN'], axis=1)

In [67]:
df.columns

Index(['TITLE', 'LOCATION', 'PRICE', 'LAND AREA', 'BUILDUP AREA',
       'ROAD ACCESS', 'FACING', 'FLOOR', 'BEDROOM', 'BATHROOM', 'BUILT YEAR',
       'PARKING', 'AMENITIES', 'PRICE_NPR', 'RATE_VALUE', 'RATE_UNIT',
       'RATE_DENOMINATOR', 'LAND_AREA_AANA'],
      dtype='object')

# now land area is done, we are moving to buildup area 

In [76]:
df['BUILDUP AREA'].head(30)
df['BUILDUP AREA'].unique()
#print(df['BUILDUP AREA'].value_counts().head(20))

array([None, '3600 Sq. Ft.', '2060 Sq. Ft.', '2500 Sq. Ft.',
       '3500 Sq. Ft.', '2870 Sq. Ft.', '-1 Sq. Ft.', '2700 Sq. Ft.',
       '2299 Sq. Ft.', '0 Sq. Ft.', '2600 Sq. Ft.', '2777.93 Sq. Ft.',
       '3000 Sq. Ft.', '4000 Sq. Ft.', '2514.5 Sq. Ft.', '1864 Sq. Ft.',
       '3800 Sq. Ft.', '3119.28 Sq. Ft.', '5600 Sq. Ft.', '1027 Sq. Ft.',
       '982 Sq. Ft.', '1096 Sq. Ft.', '3080.25 Sq. Ft.', '2940 Sq. Ft.',
       '2368.08 Sq. Ft.', '1800 Sq. Ft.', '1369 Sq. Ft.', '2738 Sq. Ft.',
       '2300 Sq. Ft.', '127 Sq. Ft.', '1500 Sq. Ft.', '4500 Sq. Ft.',
       '2250 Sq. Ft.', '1518 Sq. Ft.', '1900 Sq. Ft.', '900 Sq. Ft.',
       '2165 Sq. Ft.', '1368 Sq. Ft.', '2000 Sq. Ft.', '3303 Sq. Ft.',
       '1600 Sq. Ft.', '3026 Sq. Ft.', '1250 Sq. Ft.', '2614 Sq. Ft.',
       '2612 Sq. Ft.', '2268.16 Sq. Ft.', '5400 Sq. Ft.', '2533 Sq. Ft.',
       '9 Sq. Ft.', '6996 Sq. Ft.', '787.175 Sq. Ft.', '2800 Sq. Ft.',
       '4706 Sq. Ft.', '3200 Sq. Ft.', '1434 Sq. Ft.', '2196 Sq. Ft.',
       

# huff buildup area is a mess too

1. Multiple spellings	'Sq. Ft.', 'Sq. Feet', 'sq.ft', 'Sq ft', 'Sq.Ft'
2. Negative values	'-1 Sq. Ft.' 
3. Zero values	'0 Sq. Ft.', '0 Sq. Meter' 
4. Different units	'Sq. Meter', 'Ropani', 'ACres'
5. Weird formats	'2600+1600 Sq. Feet' (two values!)
6. Just numbers	'5', '3210', '1530', '2901' (no unit)
7. Nepali land units	'1 Ropani', '5 aana', '3 aana 2 Paisa'
8. Complex Ropani format	'0-3-2-0 Ropani-aana-Paisa-Daa

In [77]:
# Count by unit type
has_sqft = df['BUILDUP AREA'].str.contains('sq', na=False, case=False).sum()
has_feet = df['BUILDUP AREA'].str.contains('feet', na=False, case=False).sum()
has_meter = df['BUILDUP AREA'].str.contains('meter', na=False, case=False).sum()
has_ropani = df['BUILDUP AREA'].str.contains('ropani', na=False, case=False).sum()
has_aana = df['BUILDUP AREA'].str.contains('aana', na=False, case=False).sum()
has_acres = df['BUILDUP AREA'].str.contains('acres', na=False, case=False).sum()

print("BUILDUP AREA UNIT DISTRIBUTION")
print("=" * 40)
print(f"Sq. Ft/Feet: {has_sqft + has_feet}")
print(f"Sq. Meter:   {has_meter}")
print(f"Ropani:      {has_ropani}")
print(f"Aana:        {has_aana}")
print(f"Acres:       {has_acres}")
print(f"Non-null:    {df['BUILDUP AREA'].notna().sum()}")

BUILDUP AREA UNIT DISTRIBUTION
Sq. Ft/Feet: 1090
Sq. Meter:   26
Ropani:      20
Aana:        23
Acres:       1
Non-null:    719


In [78]:
# now the problamatic one
# Negative values
negative = df[df['BUILDUP AREA'].str.contains('-', na=False)]
print(f"Negative values: {len(negative)}")
print(negative['BUILDUP AREA'].tolist())

# Zero values
zero_build = df[df['BUILDUP AREA'].str.contains('^0', na=False)]
print(f"\nZero values: {len(zero_build)}")
print(zero_build['BUILDUP AREA'].tolist())

# Multiple values (has + sign)
multi = df[df['BUILDUP AREA'].str.contains(r'\+', na=False)]
print(f"\nMultiple values: {len(multi)}")
print(multi['BUILDUP AREA'].tolist())

Negative values: 22
['-1 Sq. Ft.', '3000 Ropani-aana-Paisa-Daam', '0-3-2-0 Ropani-aana-Paisa-Daam', '1784.99 Ropani-aana-Paisa-Daam', '0-11-0-0 Ropani-aana-Paisa-Daam', '0-3-2-0 Ropani-aana-Paisa-Daam', '1250 Ropani-aana-Paisa-Daam', '0-4-2-0 Ropani-aana-Paisa-Daam', '0-2-0-0 Ropani-aana-Paisa-Daam', '0-3-2-0 Ropani-aana-Paisa-Daam', '0-3-2-0 Ropani-aana-Paisa-Daam', '0-4-0-0 Ropani-aana-Paisa-Daam', '0-3-0-0 Ropani-aana-Paisa-Daam', '0-3-0-0 Ropani-aana-Paisa-Daam', '0-4-0-0 Ropani-aana-Paisa-Daam', '0-7-0-0 Ropani-aana-Paisa-Daam', '1-0-0-0 Ropani-aana-Paisa-Daam', '0-4-0-0 Ropani-aana-Paisa-Daam', '0-4-2-0 Ropani-aana-Paisa-Daam', '0-5-1-0 Ropani-aana-Paisa-Daam', '6000-6500 Sq. Meter', '0-3-3-0 Sq. Meter']

Zero values: 34
['0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Ft.', '0 Sq. Meter', '0 Sq. Meter', '0-3-2-0 Ropani-aana-Paisa-Daam', '0-

# What We're Dealing With
1. Sq. Ft/Feet	= 1,090 entries	= Easy - extract numbers
2. Sq. Meter	= 26 entries	= Convert to sq. ft (×10.764)
3. Ropani =	20 entries	= Complex format
4. Aana =	23 entries	= Convert to sq. ft
5. Acres = 1 entry	= Convert to sq. ft
6. Negative	= 22 entries	= Set to NaN (errors)
7. Zero =	34 entries	= Set to NaN (not meaningful)
8. Multiple =	1 entry	= Handle separately
9. Missing -	2,699 rows	= Keep as NaN

In [80]:
#create clean column for area sl per fit
df['BUILDUP_AREA_SQFT'] = None


In [81]:
# Mark rows with negative or zero as invalid
invalid_mask = df['BUILDUP AREA'].astype(str).str.contains('^-|^0 ', na=False)
print(f"Invalid rows (negative/zero): {invalid_mask.sum()}")
# We'll leave these as None

Invalid rows (negative/zero): 19


In [82]:
# Pattern: any number followed by 'sq' (case insensitive)
sqft_mask = df['BUILDUP AREA'].str.contains('sq', na=False, case=False)

def extract_number(value):
    if pd.isna(value):
        return None
    import re
    match = re.search(r'(\d+\.?\d*)', str(value))
    if match:
        return float(match.group(1))
    return None

df.loc[sqft_mask, 'BUILDUP_AREA_SQFT'] = df.loc[sqft_mask, 'BUILDUP AREA'].apply(extract_number)

In [83]:
# Convert sq. meter to sq. ft (× 10.764)
meter_mask = df['BUILDUP AREA'].str.contains('meter', na=False, case=False)

def convert_meter_to_sqft(value):
    if pd.isna(value):
        return None
    import re
    match = re.search(r'(\d+\.?\d*)', str(value))
    if match:
        return float(match.group(1)) * 10.764
    return None

df.loc[meter_mask, 'BUILDUP_AREA_SQFT'] = df.loc[meter_mask, 'BUILDUP AREA'].apply(convert_meter_to_sqft)

In [84]:
# 1 aana = 342.25 sq. ft
aana_mask = df['BUILDUP AREA'].str.contains('aana', na=False, case=False)

def convert_aana_to_sqft(value):
    if pd.isna(value):
        return None
    import re
    match = re.search(r'(\d+\.?\d*)', str(value))
    if match:
        return float(match.group(1)) * 342.25
    return None

df.loc[aana_mask, 'BUILDUP_AREA_SQFT'] = df.loc[aana_mask, 'BUILDUP AREA'].apply(convert_aana_to_sqft)


In [85]:
# 1 acre = 43,560 sq. ft
acres_mask = df['BUILDUP AREA'].str.contains('acres', na=False, case=False)

def convert_acres_to_sqft(value):
    if pd.isna(value):
        return None
    import re
    match = re.search(r'(\d+\.?\d*)', str(value))
    if match:
        return float(match.group(1)) * 43560
    return None

df.loc[acres_mask, 'BUILDUP_AREA_SQFT'] = df.loc[acres_mask, 'BUILDUP AREA'].apply(convert_acres_to_sqft)


In [86]:
# 1 Ropani = 64 aana = 5476 sq. ft (approx)
# Actually, let's research exact conversion later

ropani_mask = df['BUILDUP AREA'].str.contains('ropani', na=False, case=False)

def convert_ropani_to_sqft(value):
    if pd.isna(value):
        return None
    # Format: Ropani-Aana-Paisa-Daam
    # Example: "0-3-2-0" = 0 Ropani, 3 Aana, 2 Paisa, 0 Daam
    import re
    match = re.search(r'(\d+)-(\d+)-(\d+)-(\d+)', str(value))
    if match:
        ropani = float(match.group(1))
        aana = float(match.group(2))
        paisa = float(match.group(3))
        daam = float(match.group(4))
        
        # 1 Ropani = 64 Aana
        # 1 Aana = 4 Paisa
        # 1 Paisa = 4 Daam
        total_aana = (ropani * 64) + aana + (paisa / 4) + (daam / 16)
        return total_aana * 342.25  # Convert to sq. ft
    return None

df.loc[ropani_mask, 'BUILDUP_AREA_SQFT'] = df.loc[ropani_mask, 'BUILDUP AREA'].apply(convert_ropani_to_sqft)

In [87]:
#Handle Multiple Values (2600+1600 Sq. Feet)
# For the one with "+", take the sum or first value?
multi_mask = df['BUILDUP AREA'].str.contains(r'\+', na=False)

def handle_multiple(value):
    if pd.isna(value):
        return None
    import re
    numbers = re.findall(r'(\d+\.?\d*)', str(value))
    if numbers:
        return sum(float(n) for n in numbers)  # Sum both
    return None

df.loc[multi_mask, 'BUILDUP_AREA_SQFT'] = df.loc[multi_mask, 'BUILDUP AREA'].apply(handle_multiple)

In [88]:
# Remove negative and zero after conversion
df.loc[df['BUILDUP_AREA_SQFT'] <= 0, 'BUILDUP_AREA_SQFT'] = None

In [90]:

print("BUILDUP AREA CLEANING RESULTS")

print(f"Total rows with cleaned data: {df['BUILDUP_AREA_SQFT'].notna().sum()}")
print(f"Min: {df['BUILDUP_AREA_SQFT'].min():.2f} sq. ft")
print(f"Max: {df['BUILDUP_AREA_SQFT'].max():.2f} sq. ft")
print(f"Mean: {df['BUILDUP_AREA_SQFT'].mean():.2f} sq. ft")
print(f"Median: {df['BUILDUP_AREA_SQFT'].median():.2f} sq. ft")

# Check for outliers
print(f"\nRows > 10,000 sq. ft: {(df['BUILDUP_AREA_SQFT'] > 10000).sum()}")
print(f"Rows > 50,000 sq. ft: {(df['BUILDUP_AREA_SQFT'] > 50000).sum()}")

BUILDUP AREA CLEANING RESULTS
Total rows with cleaned data: 686
Min: 1.00 sq. ft
Max: 130680.00 sq. ft
Mean: 4174.92 sq. ft
Median: 2600.00 sq. ft

Rows > 10,000 sq. ft: 34
Rows > 50,000 sq. ft: 4


In [93]:
df.head()

,TITLE,LOCATION,PRICE,LAND AREA,BUILDUP AREA,ROAD ACCESS,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,None
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,None
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,None
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,None
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,None


# Buildup area is also fixed, now let's look at Road ascess

In [95]:
df['ROAD ACCESS'].unique()

array(['12 Feet', '10 Feet', '20 Feet', '13 Feet', '14 Feet', '16 Feet',
       '26 Feet', '25 Feet', '19 Feet', '18 Feet', '15 Feet', '24 Feet',
       '27 Feet', '30 Feet', '9 Meter', '22 Feet', '23 Feet', '8 Feet',
       '11 Feet', '20 Meter', None, '35 Feet', '6 Feet', '10 Meter',
       '4 Feet', '40 Feet', '17 Feet', '13 Meter', '13-20 Feet',
       '10-12 Feet', '12-18 Feet', '28 Feet', '10-15 Feet', '12-16 Feet',
       '32 Feet', '12/20 Feet', '9 Feet', '12-14 Feet', '22  Feet',
       '15-26 Feet', '14-20 Feet', '15-20 Feet', '5 Feet', '10-20 Feet',
       '12-13 Feet', '10-16 Feet', '14-26 Feet', '8-10 Feet',
       '12/13 Feet', '20  Feet', '9-12 Feet', '13  Feet', '10/13 Feet',
       '10-24 Feet', '16-22 Feet', '20-26 Feet', '13-15 Feet',
       '13-16 Feet', '12-15 Feet', '8-12 Feet', '15-24 Feet',
       '13-24 Feet', '10-26 Feet', '36 Feet', '12-20 Feet', '10-14 Feet',
       '12-26 Feet', '13-18 Feet', '15 Meter', '21 Feet', '0 Feet',
       '16-20 Feet', '10-18 Feet

# Mess
1. Ranges	'13-20 Feet', '10-12 Feet', '12-18 Feet'
2. Slash formats	'12/20 Feet', '10/13 Feet', '16/20 Feet'
3. Mixed units	'9 Meter', '20 Meter', '22 meteRs.'
4. Just numbers	'20', '13', '10', '12' (no unit!)
5. Typos	'12Feet', '22 Feet' (extra spaces)
6. Text variations	'Feet', 'FT', 'feet', "Feet's"
7. Multiple values	'12 & 22 Feet'
8. Zero	'0 Feet'

In [97]:
# since it is in ony feet and meters i hope it wont be as hard as others we did before
df['ROAD_ACCESS_FEET'] = None

In [98]:
def extract_single_feet(value):
    if pd.isna(value):
        return None
    import re
    # Look for number followed by Feet/FT (case insensitive)
    match = re.search(r'(\d+\.?\d*)\s*(?:Feet|FT|feet|ft)', str(value), re.IGNORECASE)
    if match:
        return float(match.group(1))
    return None

simple_mask = df['ROAD ACCESS'].str.contains(r'\d+\s*(?:Feet|FT|feet|ft)', na=False, case=False)
df.loc[simple_mask, 'ROAD_ACCESS_FEET'] = df.loc[simple_mask, 'ROAD ACCESS'].apply(extract_single_feet)

In [99]:
# Convert meters to feet (× 3.28084)
def convert_meter_to_feet(value):
    if pd.isna(value):
        return None
    import re
    match = re.search(r'(\d+\.?\d*)\s*(?:Meter|meteRs|M)', str(value), re.IGNORECASE)
    if match:
        return float(match.group(1)) * 3.28084
    return None

meter_mask = df['ROAD ACCESS'].str.contains('meter', na=False, case=False)
df.loc[meter_mask, 'ROAD_ACCESS_FEET'] = df.loc[meter_mask, 'ROAD ACCESS'].apply(convert_meter_to_feet)

In [100]:
# If value is just a number (like '20', '13', '10')
def extract_plain_number(value):
    if pd.isna(value):
        return None
    import re
    # Check if string is just number (no letters)
    if re.match(r'^\d+\.?\d*$', str(value).strip()):
        return float(value)
    return None

plain_mask = df['ROAD ACCESS'].astype(str).str.match(r'^\d+\.?\d*$', na=False)
df.loc[plain_mask, 'ROAD_ACCESS_FEET'] = df.loc[plain_mask, 'ROAD ACCESS'].apply(extract_plain_number)

In [101]:
# For '13-20 Feet' - what to do?
# Take average (16.5) or minimum (13)

def extract_range_average(value):
    if pd.isna(value):
        return None
    import re
    # Look for pattern like '13-20'
    match = re.search(r'(\d+\.?\d*)\s*-\s*(\d+\.?\d*)', str(value))
    if match:
        low = float(match.group(1))
        high = float(match.group(2))
        return (low + high) / 2  # Average
    return None

range_mask = df['ROAD ACCESS'].str.contains(r'\d+\.?\d*\s*-\s*\d+\.?\d*', na=False)
df.loc[range_mask, 'ROAD_ACCESS_FEET'] = df.loc[range_mask, 'ROAD ACCESS'].apply(extract_range_average)

In [102]:
# For '12/20 Feet' - treat same as range
slash_mask = df['ROAD ACCESS'].str.contains(r'\d+\.?\d*/\d+\.?\d*', na=False)
df.loc[slash_mask, 'ROAD_ACCESS_FEET'] = df.loc[slash_mask, 'ROAD ACCESS'].apply(extract_range_average)


In [103]:
# For '12 & 22 Feet'
amp_mask = df['ROAD ACCESS'].str.contains(r'&', na=False)
df.loc[amp_mask, 'ROAD_ACCESS_FEET'] = df.loc[amp_mask, 'ROAD ACCESS'].apply(extract_range_average)

In [104]:
df.loc[df['ROAD_ACCESS_FEET'] <= 0, 'ROAD_ACCESS_FEET'] = None


In [105]:
df.head()

,TITLE,LOCATION,PRICE,LAND AREA,BUILDUP AREA,ROAD ACCESS,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT,ROAD_ACCESS_FEET
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,None,12.0
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,None,10.0
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,None,10.0
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,None,12.0
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,None,20.0


In [112]:
# Rename original for reference
df = df.rename(columns={'ROAD ACCESS': 'ROAD_ACCESS_RAW'})
df = df.rename(columns={'LAND AREA': 'LAND_AREA_RAW'})
df = df.rename(columns={'BUILDUP AREA': 'BUILDUP_AREA_RAW'})
df = df.rename(columns={'PRICE': 'PRICE_RAW'})


In [113]:
df.head()

,TITLE,LOCATION,PRICE_RAW,LAND_AREA_RAW,BUILDUP_AREA_RAW,ROAD_ACCESS_RAW,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT,ROAD_ACCESS_FEET
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,None,12.0
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,None,10.0
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,None,10.0
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,None,12.0
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,None,20.0


In [115]:
# Convert to numeric (coerce errors to NaN)
df['BUILDUP_AREA_SQFT'] = pd.to_numeric(df['BUILDUP_AREA_SQFT'], errors='coerce')
df['ROAD_ACCESS_FEET'] = pd.to_numeric(df['ROAD_ACCESS_FEET'], errors='coerce')



In [117]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3418 entries, 0 to 3417
Data columns (total 20 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   TITLE              3418 non-null   object 
 1   LOCATION           3418 non-null   object 
 2   PRICE_RAW          3418 non-null   object 
 3   LAND_AREA_RAW      3329 non-null   object 
 4   BUILDUP_AREA_RAW   719 non-null    object 
 5   ROAD_ACCESS_RAW    3409 non-null   object 
 6   FACING             3212 non-null   object 
 7   FLOOR              3323 non-null   float64
 8   BEDROOM            3136 non-null   float64
 9   BATHROOM           3072 non-null   float64
 10  BUILT YEAR         3357 non-null   object 
 11  PARKING            632 non-null    object 
 12  AMENITIES          3418 non-null   object 
 13  PRICE_NPR          2667 non-null   float64
 14  RATE_VALUE         560 non-null    float64
 15  RATE_UNIT          560 non-null    object 
 16  RATE_DENOMINATOR   560 n

In [118]:
df.to_parquet('./data/house_price_nepal_final_cleaned.parquet')
df.to_csv('./data/house_price_nepal_final_cleaned.csv', index=False)

In [119]:
# huff, finally the cleanup finished

In [120]:
df.head()

,TITLE,LOCATION,PRICE_RAW,LAND_AREA_RAW,BUILDUP_AREA_RAW,ROAD_ACCESS_RAW,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT,ROAD_ACCESS_FEET
0,House for Sale,"Imadol, Lalitpur",Rs. 2.9 Cr,4.0 aana,None,12 Feet,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,NaN,12.0
1,House for Sale,"Satdobato, Lalitpur",Rs. 4.75 Cr,3.0 aana,None,10 Feet,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,NaN,10.0
2,4 BHK House for Sale,"Imadol, Lalitpur",Rs. 1.99 Cr,2.3 aana,None,10 Feet,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,NaN,10.0
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",Rs. 4 Cr,7.0 aana,None,12 Feet,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,NaN,12.0
4,House for Rent,"Maharajgunj, Kathmandu",Rs. 12000000,6.0 aana,None,20 Feet,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,NaN,20.0


In [122]:
df.head()

,TITLE,LOCATION,LAND_AREA_RAW,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT,ROAD_ACCESS_FEET
0,House for Sale,"Imadol, Lalitpur",4.0 aana,West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,NaN,12.0
1,House for Sale,"Satdobato, Lalitpur",3.0 aana,West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,NaN,10.0
2,4 BHK House for Sale,"Imadol, Lalitpur",2.3 aana,West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,NaN,10.0
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",7.0 aana,North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,NaN,12.0
4,House for Rent,"Maharajgunj, Kathmandu",6.0 aana,South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,NaN,20.0


In [123]:
df.to_parquet('./data/house_price_nepal_final_cleaned.parquet')
df.to_csv('./data/house_price_nepal_final_cleaned.csv', index=False)

In [126]:
df = df.drop(['LAND_AREA_RAW'], axis=1)

In [127]:
df.head()

,TITLE,LOCATION,FACING,FLOOR,BEDROOM,BATHROOM,BUILT YEAR,PARKING,AMENITIES,PRICE_NPR,RATE_VALUE,RATE_UNIT,RATE_DENOMINATOR,LAND_AREA_AANA,BUILDUP_AREA_SQFT,ROAD_ACCESS_FEET
0,House for Sale,"Imadol, Lalitpur",West,3.0,5.0,4.0,2076 B.S,1 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",29000000.0,NaN,None,None,4.0,NaN,12.0
1,House for Sale,"Satdobato, Lalitpur",West,4.5,5.0,6.0,2076 B.S,2 CaRs. & 2 Bikes,"['Earthquake Resistant', 'Parquet', 'Drinking ...",47500000.0,NaN,None,None,3.0,NaN,10.0
2,4 BHK House for Sale,"Imadol, Lalitpur",West,2.5,4.0,4.0,2060 B.S,1 CaRs. & 3 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",19900000.0,NaN,None,None,2.3,NaN,10.0
3,Bungalow House for Sale,"Bhaisepati, Lalitpur",North-West,2.5,4.0,3.0,2059 B.S,4 CaRs. & 4 Bikes,"['Earthquake Resistant', 'Marbel', 'Parquet', ...",40000000.0,NaN,None,None,7.0,NaN,12.0
4,House for Rent,"Maharajgunj, Kathmandu",South,2.0,4.0,4.0,2071 B.S,4 CaRs. & 5 Bikes,"['Earthquake Resistant', 'Parquet', 'Parking',...",12000000.0,NaN,None,None,6.0,NaN,20.0


In [128]:
df.to_parquet('./data/house_price_nepal_final_cleaned.parquet')
df.to_csv('./data/house_price_nepal_final_cleaned.csv', index=False)